docs/rough_edges.md

```diff
@@ -36,3 +36,11 @@ v2.
 
   **Workaround**: use `ClientCapabilities.RootsV2`, which aligns with the
   semantics of other capability fields.
+
+- Default capabilities should have been empty. Instead, servers default to
+  advertising `logging`, and clients default to advertising `roots` with
+  `listChanged: true`. This is confusing because a nil `Capabilities` field
+  does not mean "no capabilities".
+
+  **Workaround**: set `ServerOptions.Capabilities` or `ClientOptions.Capabilities`
```

Comment:
IMO it would be slightly clearer to be very explicit here about the difference between an empty Capabilities and a `nil` capabilities.

I'd reword here to: "**Workaround**: to advertise no capabilities set `ServerOptions`... to an empty ..."